examples/workflows/unet_training_dict.py

```diff
@@ -162,7 +162,7 @@ def main():
         optimizer=opt,
         loss_function=loss,
         inferer=SimpleInferer(),
-        amp=False,
+        amp=True,  # if no FP16 support in GPU or PyTorch version < 1.6, change to False
```

Comment:
Can this value be set automatically with introspection?

Reply:
could check the pytorch version with https://github.com/Project-MONAI/MONAI/blob/3a47c6ee294846e65753ccd815abdedb8eff4eb8/monai/transforms/spatial/array.py#L51

Reply:
Hi @hjmjohnson and @wyli ,

PyTorch version is available to parse.
Do you know how to detect whether the GPU in use can support FP16?
I don't have old type GPU besides, do you have some experience on that?
Thanks.

Reply:
will pytorch raise some exceptions with incompatible gpus?

Reply:
> will pytorch raise some exceptions with incompatible gpus?

I don't know, could you please help test on some old GPU?
At least, I think if `amp=True` on an old GPU can't achieve the expected result, so I highlighted in the comment.

Thanks.

Reply:
> > will pytorch raise some exceptions with incompatible gpus?
> 
> I don't know, could you please help test on some old GPU?
> At least, I think if `amp=True` on an old GPU can't achieve the expected result, so I highlighted in the comment.
> 
> Thanks.

sure I can test it with titan xp when I have time

Reply:
Thanks @wyli , I will change to use `amp=True if if get_torch_version_tuple() >= (1, 6) else False` in this example.

Reply:
updated in the latest commit.
Thanks.